src/PathProcessor/CivicrmPathProcessor.php

```diff
@@ -11,43 +11,40 @@
  */
 class CivicrmPathProcessor implements InboundPathProcessorInterface {
 
+  private static $knownPaths;
+
   /**
    * {@inheritdoc}
    */
   public function processInbound($path, Request $request) {
     // If the path is a civicrm path.
     if (strpos($path, '/civicrm/') === 0) {
+      // We've already looked up this path on this page request.
+      if (isset(self::$knownPaths[$path])) {
+        return self::$knownPaths[$path];
+      }
       // Initialize civicrm.
       $civicrm = new Civicrm();
       $civicrm->initialize();
-      // Fetch civicrm menu items.
-      $items = \CRM_Core_Menu::items();
-      $longest = '';
-      foreach (array_keys($items) as $item) {
-        $item = '/' . $item;
-        // If he current path is a civicrm path.
-        if ((strpos($path, $item) === 0)) {
-          // Discover longest matching civicrm path in the request path.
-          if (strlen($item) > strlen($longest)) {
-            $longest = $item;
-          }
-        }
+      $routesInitialized = \CRM_Core_DAO::singleValueQuery('SELECT COUNT(*) from civicrm_menu');
+      if (!$routesInitialized) {
+        \CRM_Core_Menu::store();
```

Comment:
The test issue is that during an initial install of civi, this line generates an infinite loop, because there is nothing in civicrm_menu, and then calling store() ends up back here again recursively while trying to build menu links.

That might also happen during a cache clear - I didn't test that.